### PR TITLE
feat: allow overrides on add-on values at install time

### PR DESCRIPTION
**Pull request by ricardomaraschini**
when installing take the user provided overrides for all add-ons.

### CODE REVIEW

File: pkg/addons/adminconsole/adminconsole.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func (a *AdminConsole) Version() (map[string]string, error) {
 }
 
 func (a *AdminConsole) Name() string {
-	return "AdminConsole"
+	return releaseName
 }
```
Comment (laverya) [inline pkg/addons/adminconsole/adminconsole.go:114]: I have a feeling this is going to break something, let's see the version test

File: pkg/addons/velero/velero.go [hunk 2]
```diff
@@ -135,7 +135,11 @@ func (o *Velero) Outro(ctx context.Context, cli client.Client) error {
 	return nil
 }
 
+type Options struct {
+	IsEnabled bool
+}
```
Comment (laverya) [inline pkg/addons/velero/velero.go:140]: why the change to pass an options struct? personally I want to have to fix the compile errors for each call to 'new' when I add a new field, as it's almost always going to be required

File: pkg/addons/applier.go [hunk 3]
```diff
@@ -24,15 +22,14 @@ import (
...
-	"github.com/replicatedhq/embedded-cluster/pkg/spinner"
 )
 
 // AddOn is the interface that all addons must implement.
 type AddOn interface {
 	Version() (map[string]string, error)
 	Name() string
 	HostPreflights() (*v1beta2.HostPreflightSpec, error)
-	GenerateHelmConfig(onlyDefaults bool) ([]v1beta1.Chart, []v1beta1.Repository, error)
+	GenerateHelmConfig(onlyDefaults bool) (*v1beta1.Chart, *v1beta1.Repository, error)
```
Comment (laverya) [inline pkg/addons/applier.go:32]: you seem to be doing this in order to allow overriding helm configs by addon name, but the operator code (that's been merged already) overrides helm configs by chart name - I would greatly prefer if we just overrode configs by chart name everywhere

File: pkg/addons/adminconsole/adminconsole.go (hunk 1 above)
Comment (laverya) [reply pkg/addons/adminconsole/adminconsole.go:114]: surprisingly it didn't break unit tests!

Comment (ricardomaraschini) [reply pkg/addons/adminconsole/adminconsole.go]: restored anyways.

File: pkg/addons/velero/velero.go (hunk 2 above)
Comment (ricardomaraschini) [reply pkg/addons/velero/velero.go]: restored the original arguments.

File: pkg/addons/applier.go (hunk 3 above)
Comment (ricardomaraschini) [reply pkg/addons/applier.go]: fixed.

**Review by laverya — APPROVED** [review]
